spacy_stanza/tokenizer.py

```diff
@@ -1,124 +1,60 @@
-# coding: utf8
-from spacy.language import Language
-from spacy.tokens import Doc, Span
-from spacy.util import get_lang_class
-from spacy.vocab import create_vocab
-
-from stanza.models.common.vocab import UNK_ID
-from stanza.models.common.pretrain import Pretrain
-from stanza import Document
-
-import numpy
-import re
 import warnings
+from typing import Optional, Union
 
+from spacy.tokens import Doc
+from spacy.util import registry
 
-class StanzaLanguage(Language):
-    def __init__(self, snlp, meta=None, **kwargs):
-        """Initialize the Language class.
+from stanza import Pipeline
+from stanza.models.common.vocab import UNK_ID
+from stanza.models.common.pretrain import Pretrain
 
-        Instead of "en" etc. we call the language "stanza_en" to not
-        cause conflicts with spaCy's built-in languages. Using entry points,
-        this also allows serializing and deserializing the language class
-        and "lang": "stanza_en" in the meta.json will automatically
-        instantiate this class if this package is available.
 
-        snlp (stanza.Pipeline): The loaded Stanza pipeline.
-        kwargs: Optional config parameters.
-        RETURNS (spacy.language.Language): The nlp object.
-        """
-        if hasattr(snlp, "lang"):
-            lang = snlp.lang
-        else:
-            # backward compatible with stanza v1.0.0
-            lang = snlp.processors["tokenize"].config["lang"]
-        self.snlp = snlp
-        self.svecs = StanzaLanguage._find_embeddings(snlp)
-        self.lang = "stanza_" + lang
-        self.Defaults = get_defaults(lang)
-        self.vocab = create_vocab(lang, self.Defaults)
-        self.tokenizer = Tokenizer(snlp, self.vocab)
-        self._components = []
-        self._disabled = set()
-        self.max_length = kwargs.get("max_length", 10 ** 6)
-        self.batch_size = kwargs.get("batch_size", 256)
-        self._meta = (
-            {"lang": self.lang, "stanza": snlp.config} if meta is None else dict(meta)
+@registry.tokenizers("spacy_stanza.PipelineAsTokenizer.v1")
+def create_tokenizer(
+    lang: str = "",
+    model_dir: str = "",
+    package: str = "default",
+    processors: Union[dict, str] = {},
+    logging_level: Optional[Union[int, str]] = None,
+    verbose: Optional[bool] = None,
+    use_gpu: bool = True,
+    kwargs: dict = {},
+):
+    def tokenizer_factory(
+        nlp,
+        lang=lang,
+        model_dir=model_dir,
+        package=package,
+        processors=processors,
+        logging_level=logging_level,
+        verbose=verbose,
+        use_gpu=use_gpu,
+        kwargs=kwargs,
+    ) -> StanzaTokenizer:
+        snlp = Pipeline(
+            lang=lang,
+            model_dir=model_dir,
```

Comment:
Hi, I was just testing this PR, and found that `"model_dir"` is not the correct name for this parameter in Stanza. It should be just `"dir"` (see https://github.com/stanfordnlp/stanza/blob/9aa915e374b8f362c554e53108ab8974a774e408/stanza/pipeline/core.py#L74). It doesn't currently fail if you use the default directory since extra kwargs are being swallowed silently by Stanza's Pipeline.

I don't know if you want to just change it here, when passing it to Stanza, or all the way up the spacy-stanza API (the two occurrences above). I'd imagine the latter to keep the API in sync with Stanza?

Reply:
Thanks for the note! I don't like using `dir` as a variable name in general, but here the goal is to mirror the stanza API.